src/algebra/big_operators/basic.lean

```diff
@@ -549,6 +549,24 @@ begin
   exact ⟨x, (mem_filter.1 hx).1, (mem_filter.1 hx).2⟩
 end
 
+@[to_additive]
+lemma prod_subset_one_on_sdiff (h : s₁ ⊆ s₂) (hg : ∀ x ∈ s₂, x ∉ s₁ → g x = 1)
+  (hfg : ∀ x ∈ s₁, f x = g x) : ∏ i in s₁, f i = ∏ i in s₂, g i :=
```

Comment:
Would it be easier if you used `sdiff` in the statement?

Reply:
Good point. I've made the change. There's one thing I'm not sure of though. To get Lean to accept `s₁ \ s₂` in the statement of the lemma I had to add `[decidable_eq α]` to the hypotheses. Is that okay?

Reply:
The linter does not complain so I am happy. (The rule is that a lemma/theorem should only take a `decidable_eq` argument if it is used elsewhere in the type.)